ethereum/contracts/nft/NFTBridge.sol

```diff
@@ -233,19 +236,18 @@ contract NFTBridge is NFTBridgeGovernance {
         transfer.tokenID = encoded.toUint256(index);
         index += 32;
 
-        uint8 len_uri = encoded.toUint8(index);
-        index += 1;
+        // From here we read backwards due malformatted package
+        index = encoded.length;
 
-        transfer.uri = string(encoded.slice(index, len_uri));
-        index += len_uri;
+        index -= 2;
+        transfer.toChain = encoded.toUint16(index);
 
+        index -= 32;
         transfer.to = encoded.toBytes32(index);
-        index += 32;
 
-        transfer.toChain = encoded.toUint16(index);
-        index += 2;
+        transfer.uri = string(encoded.slice(index, encoded.length - index));
```

Comment:
this would return `0x00000000000000000000000017a77d63765963d0cb9dec12fa8f62e68fee8fd40005` in the case of the bad payload. 34 from the end to the end essentially. Instead, see earlier comment.